cmd/habitatctl/commands/sources.go

```diff
@@ -0,0 +1,158 @@
+package commands
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/eagraf/habitat/cmd/sources"
+	"github.com/eagraf/habitat/pkg/compass"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+)
+
+// communityCmd represents the community command
+var sourcesCmd = &cobra.Command{
+	Use:   "sources",
+	Short: "Habitat sources is the data access layer on top of JSON, relational, and key-value data.",
+	Long: `Subcommands:
+	read <source_file>
+	join <source_file>
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+	},
+}
+
+var sourcesReadCmd = &cobra.Command{
+	Use:   "read",
+	Short: "read a source file",
+	Run: func(cmd *cobra.Command, args []string) {
+
+		id := cmd.Flags().Lookup("id").Value.String()
+		path := cmd.Flags().Lookup("path").Value.String()
+
+		fmt.Printf("Source Read Request for %s\n", id)
+
+		req := &sources.ReadRequest{
+			SourceName: sources.SourceName(id),
+		}
+
+		reader := sources.NewReader(sources.NewJSONReader(cmd.Context(), path), sources.NewBasicPermissionsManager())
+
+		allowed, err, data := reader.Read(req)
+
+		if err != nil {
+			log.Error().Msgf("Error reading source: %s", err.Error())
+		} else {
+			fmt.Printf("Allowed: %t, Data: %s\n", allowed, data)
+		}
+	},
+}
+
+var sourcesWriteCmd = &cobra.Command{
+	Use:   "write",
+	Short: "write a source file",
+	Run: func(cmd *cobra.Command, args []string) {
+
+		id := cmd.Flags().Lookup("id").Value.String()
+		path := cmd.Flags().Lookup("path").Value.String()
+		data := cmd.Flags().Lookup("data").Value.String()
+
+		fmt.Printf("Source Write Request for %s, %s\n", id, data)
+
+		req := &sources.WriteRequest{
+			SourceName: sources.SourceName(id),
+			Data:       json.RawMessage(data),
+		}
+
+		writer := sources.NewWriter(sources.NewJSONWriter(cmd.Context(), path), sources.NewBasicPermissionsManager())
```

Comment:
Shouldn't we access a sources server over the internet rather than directly to local disk? `habitatctl` is mainly just a client, so this could be running on any machine, not necessarily a Habitat node